src/Sulu/Bundle/TestBundle/Testing/SetDomainEventUserSubscriber.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\TestBundle\Testing;
+
+use Prophecy\Prophecy\ProphecySubjectInterface;
+use Sulu\Bundle\EventLogBundle\Application\Subscriber\SetDomainEventUserSubscriber as BaseSetDomainEventUserSubscriber;
+use Sulu\Bundle\EventLogBundle\Domain\Event\DomainEvent;
+
+class SetDomainEventUserSubscriber extends BaseSetDomainEventUserSubscriber
+{
+    public function setDomainEventUser(DomainEvent $event): void
+    {
+        parent::setDomainEventUser($event);
+
+        // Unset user, if it's a prophesized double
+        if ($event->getUser() instanceof ProphecySubjectInterface) {
+            $event->setUser(null);
```

Comment:
@nnatter @alexander-schranz There are some tests prophesizing the `UserInterface` and setting it on the `TokenStorage` ... This results in failing tests, because doctrine of course cannot store that prophecy in the database. Do you think it's a good idea to just override the `SetDomainEventUserSubscriber` in the `SuluTestBundle`?

The failing tests: https://github.com/sulu/sulu/pull/5947/checks?check_run_id=2402130904

Reply:
I would change this test cases. This should not happened that functional tests are using `prophesize` or any other mocks.

Reply:
fixed